### PR TITLE
Add HP33120A burst modulation parameters

### PR DESCRIPTION
**Pull request by mzen228**
Add support for burst modulation for the HP33120A arbitrary waveform generator

### CODE REVIEW

File: pymeasure/instruments/hp/hp33120A.py [hunk 1]
```diff
@@ -112,6 +112,54 @@ def __init__(self, adapter, name="Hewlett Packard 33120A Function Generator", **
         values=AMPLITUDE_UNITS,
         map_values=True
     )
+    burst_state = Instrument.control(
+        "BM:STATE?", "BM:STATE %d",
+        """Enable or disable burst modulation""",
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp33120A.py:117]: It's better to call it `burst_enabled`, which makes it obvious, what True or False means: `burst_enabled = True`.

File: pymeasure/instruments/hp/hp33120A.py [hunk 2]
```diff
@@ -112,6 +112,54 @@ def __init__(self, adapter, name="Hewlett Packard 33120A Function Generator", **
...
+        validator=strict_discrete_set,
+        values=[0, 1],
+    )
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp33120A.py:120]: You can use the `map_values` function to translate True/False to/from 0,1:
```suggestion
        values={True: 1, False: 0},
        map_values=True,
    )
```

File: pymeasure/instruments/hp/hp33120A.py [hunk 3]
```diff
@@ -112,6 +112,54 @@ def __init__(self, adapter, name="Hewlett Packard 33120A Function Generator", **
...
+    burst_source = Instrument.control(
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp33120A.py:121]: Please add a new line between two definitions.

File: pymeasure/instruments/hp/hp33120A.py [hunk 4]
```diff
@@ -112,6 +112,54 @@ def __init__(self, adapter, name="Hewlett Packard 33120A Function Generator", **
...
+        "BM:SOURCE?", "BM:SOURCE %s",
+        """Select an internal or external gate source for burst modulation""",
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp33120A.py:123]: A `control`'s docstring should start with "Control". Similarly, a `measurement`'s docstring should start with "Get" or "Measure" and a `setting`'s docstring with "Set".
Please adjust all your docstrings.

Maybe other properties do not adhere to these conventions, but new properties should follow them.

**Review by BenediktBurger — COMMENTED** [review]
Hey @mzen228,
thank you very much for your contribution.

I marked some general comments.

You may add yourself to the `authors.txt` as well.

Consider writing [protocol tests](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#protocol-tests) for your new methods, in order to ensure, that they do, what you expect them to do.

File: pymeasure/instruments/hp/hp33120A.py (hunk 1 above)
Comment (mzen228) [reply pymeasure/instruments/hp/hp33120A.py]: resolved in [900d4a1](https://github.com/pymeasure/pymeasure/pull/1056/commits/900d4a112c5216f07a61a9f97cf3e66c791424df)

File: pymeasure/instruments/hp/hp33120A.py (hunk 2 above)
Comment (mzen228) [reply pymeasure/instruments/hp/hp33120A.py]: resolved in [900d4a1](https://github.com/pymeasure/pymeasure/pull/1056/commits/900d4a112c5216f07a61a9f97cf3e66c791424df)

File: pymeasure/instruments/hp/hp33120A.py (hunk 3 above)
Comment (mzen228) [reply pymeasure/instruments/hp/hp33120A.py:134]: resolved in [900d4a1](https://github.com/pymeasure/pymeasure/pull/1056/commits/900d4a112c5216f07a61a9f97cf3e66c791424df)

File: pymeasure/instruments/hp/hp33120A.py [hunk 5]
```diff
@@ -113,6 +123,67 @@ def __init__(self, adapter, name="Hewlett Packard 33120A Function Generator", **
...
+        "BM:PHAS? MIN",
+        """Get the minimum :attr:`~.HP33120A.burst_phase`"""
+    )
+
+    max_burst_phase = Instrument.measurement(
+        "BM:PHAS? MAX",
+        """Get the maximum :attr:`~.HP33120A.burst_phase`"""
+    )
+
+
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp33120A.py:186]: Here you have too many blank lines: 2 instead of 1, which causes one check to fail.
```suggestion
```

**Review by BenediktBurger — APPROVED** [review]
One last change due to a superfluous line.

If you are ready, I'll merge the code after that change.

**Comment by mzen228** [comment]
@BenediktBurger 
thank you for your feedback.
should be all ready now.

File: tests/instruments/hp/test_hp33120A.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        inst.burst_enabled = True
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:STATE?', b'1')],
+     True),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:STATE?', b'1')],
+     True),
```
Comment (BenediktBurger) [inline tests/instruments/hp/test_hp33120A.py:99]: I guess, you want to test both cases: true and false.
Here you test true two times!

Comment (mzen228) [reply tests/instruments/hp/test_hp33120A.py]: thank you for catching this.
resolved in https://github.com/pymeasure/pymeasure/pull/1056/commits/ffc63b0b4b04c0de094fa1465dcd5d18aae1c0c2

**Comment by BenediktBurger** [comment]
Thanks for your contribution.

**Pull request merged**